ZEngine/src/Attachment.cpp

```diff
@@ -0,0 +1,115 @@
+#include <Rendering/Renderers/RenderPasses/Attachment.h>
+#include <Hardwares/VulkanDevice.h>
+
+using namespace ZEngine::Hardwares;
+using namespace ZEngine::Rendering::Specifications;
+
+namespace ZEngine::Rendering::Renderers::RenderPasses
+{
+    Attachment::Attachment(const Specifications::AttachmentSpecification& spec)
+    {
+        ZENGINE_VALIDATE_ASSERT(!spec.ColorsMap.empty(), "Color attachments can't be empty")
+
+        VkSubpassDescription                 subpass_description                   = {};
+        std::vector<VkAttachmentDescription> attachment_description_collection     = {};
+        VkAttachmentReference                depth_attachment_reference            = {.attachment = UINT32_MAX, .layout = VK_IMAGE_LAYOUT_UNDEFINED};
```

Comment:
Could use VK_ATTACHMENT_UNUSED, which is the same as UINT32_MAX incidentially, but more in line with the API.